Description: Try to load native library first from /usr/lib/jni
 instead of extractDir or classpath.
Author: Damien Raude-Morvan <drazzib@debian.org>
Last-Update: 2013-01-12
Forwarded: no
--- a/src/main/java/net/rubygrapefruit/platform/internal/NativeLibraryLocator.java
+++ b/src/main/java/net/rubygrapefruit/platform/internal/NativeLibraryLocator.java
@@ -31,6 +31,13 @@
     }
 
     public File find(String libraryFileName) throws IOException {
+        // Try to load /usr/lib/jni/libnative-platform.so first
+        // before extractDir or classpath
+        File usrLibFile = new File("/usr/lib/jni/" + libraryFileName);
+        if (usrLibFile.isFile()) {
+            return usrLibFile;
+        }
+
         if (extractDir != null) {
             File libFile = new File(extractDir, String.format("%s/%s", NativeLibraryFunctions.VERSION, libraryFileName));
             File lockFile = new File(libFile.getParentFile(), libFile.getName() + ".lock");
